flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/JobSubmitTest.java

```diff
@@ -170,4 +170,43 @@ public void initializeOnMaster(ClassLoader loader) throws Exception {
 			fail(e.getMessage());
 		}
 	}
+
+	/**
+	 * Verifies failure when the client and job manager versions mismatch
+	 */
+	@Test
+	public void testFailureClientJobManagerVersionMismatch() {
+		try {
+			// create a simple job graph
+
+			JobVertex jobVertex = new JobVertex("Vertex that fails in initializeOnMaster") {
+
+				@Override
+				public void initializeOnMaster(ClassLoader loader) throws Exception {
+					throw new RuntimeException("test exception");
```

Comment:
Why using a faulty `JobVertex` here. This might also be the reason for the "expected" `JobExecutionException` then. And you don't check properly whether that's not the cause for it.